src/MyPlot/provider/DataProvider.php

```diff
@@ -40,7 +47,7 @@ public function __construct(MyPlot $plugin, int $cacheSize = 0) {
 	 * @param $Z
 	 * @return Plot|null
 	 */
-	protected final function getPlotFromCache(string $levelName, int $X, int $Z) {
+	protected final function getPlotFromCache(string $levelName, int $X, int $Z) : ?Plot {
```

Comment:
Because i can not a review on https://github.com/jasonwynn10/MyPlot/pull/103/files#diff-6a73d80afd1ae87787f01bb47d36361eR37, here:
That cache function is useless. It caches a plot, adds it at the END of the array.
Then you use array POP, which removes from the END of the array. So the cached plots always remain the same.
array_shift would be the right way to go with, caching only the latest cached plots.

Reply:
I didn't notice that before :man_shrugging:

Reply:
var_dumping only the plot-id's will confirm it ;)